test/functionalTest/cases/0987_put_attributes/put_entity_error.test

```diff
@@ -0,0 +1,153 @@
+# Copyright 2015 Telefonica Investigacion y Desarrollo, S.A.U
+#
+# This file is part of Orion Context Broker.
+#
+# Orion Context Broker is free software: you can redistribute it and/or
+# modify it under the terms of the GNU Affero General Public License as
+# published by the Free Software Foundation, either version 3 of the
+# License, or (at your option) any later version.
+#
+# Orion Context Broker is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU Affero
+# General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with Orion Context Broker. If not, see http://www.gnu.org/licenses/.
+#
+# For those usages not covered by this license please contact with
+# iot_support at tid dot es
+
+# VALGRIND_READY - to mark the test ready for valgrindTestSuite.sh
+
+--NAME--
+PUT /v2/entities/E1 on entity that doesn't exist
+
+--SHELL-INIT--
+dbInit CB
+brokerStart CB
+
+--SHELL--
+
+#
+# 01. Modify non-existing entity E1
+# 02. Make sure E1 does not exist
+# 03. Create E1
+# 04. Make sure E1 is as created
+# 05. Modify E1
+# 06. Make sure E1 is as modified
+#
+
+echo "01. Modify non-existing entity E1"
+echo "================================="
+payload='{ "attr1": 1 }'
+orionCurl --url /v2/entities/E1 --payload "$payload" --json -X PUT
+echo
+echo
+
+
+echo "02. Make sure E1 does not exist"
+echo "==============================="
+orionCurl --url /v2/entities/E1 --json
+echo
+echo
+
+
+echo "03. Create E1"
+echo "============="
+payload='{ "attr1": 3 }'
+orionCurl --url /v2/entities/E1 --payload "$payload" --json
+echo
+echo
+
+
+echo "04. Make sure E1 is as created"
+echo "=============================="
+orionCurl --url /v2/entities/E1 --json
+echo
+echo
+
+
+echo "05. Modify E1"
+echo "============="
```

Comment:
And not only functests ... this doesn't work. The broker doesn't accept 'unknown' attributes, as it seems. A bigger change will be necessary in mongoBackend, I'm afraid ... :-(
Also, we need to not break API V1 at the same time. Might get complicated ...